src/handwritten/_install_daqmx.py

```diff
@@ -0,0 +1,185 @@
+_META_DATA_URL = "http://localhost:8000/ni-daqmx-update-metadata.json"
+
+
+def _is_admin():
+    """
+    Verify whether the user has Administrator privileges.
+    """
+    try:
+        return ctypes.windll.shell32.IsUserAnAdmin()
+    except Exception as e:
+        print(f"Error checking admin privileges: {e}")
+        return False
+
+
+def _parse_version(version):
+    """
+    Split the version string into an array of integers.
+    """
+    return [int(part) for part in version.split(".")]
+
+
+def _compare_version(installed_ver, source_ver):
+    """
+    Compare two versions which are in the format: major.minor.update
+    """
+    iver_parts = _parse_version(installed_ver)
+    sver_parts = _parse_version(source_ver)
+
+    for iver_num, sver_num in zip(iver_parts + [0], sver_parts + [0]):
+        if iver_num < sver_num:
+            return -1
+        elif iver_num > sver_num:
+            return 1
+
+    return 0
+
+
+def _check_daqmx_is_installed():
+    """
+    Check for existing installation of DAQmx by checking for Version information.
+    Version information is read from the registry key.
+    """
+    try:
+        with winreg.OpenKeyEx(
+            winreg.HKEY_LOCAL_MACHINE,
+            r"SOFTWARE\National Instruments\NI-DAQmx\CurrentVersion",
+            0,
+            winreg.KEY_READ | winreg.KEY_WOW64_32KEY,
+        ) as daqmx_reg_key:
+            product_name = winreg.QueryValueEx(daqmx_reg_key, "ProductName")[0]
+            product_version = winreg.QueryValueEx(daqmx_reg_key, "Version")[0]
+
+        if product_name == "NI-DAQmx":
+            return product_version
+    except Exception as e:
+        print(f"Info: No existing NI-DAQmx installation found")
+
+    return False
+
+
+def _get_temp_file_name():
+    """
+    Generate a temporary file name.
+    """
+    try:
+        _, temp_file_path = tempfile.mkstemp()
+        return temp_file_path + "_daq-mx-online.exe"
+    except Exception as e:
+        print(f"error: Unable to create temporary file: {e}")
+        return None
+
+
+def _get_url_and_version_for_windows_installer(metadata_url):
+    """
+    Parse the JSON data and retrieve the download link and version information.
+    """
+    print("Log1")
+    try:
+        with urllib.request.urlopen(metadata_url) as url:
+            data = json.load(url)
+            windows_os_meta = data.get("Windows", [])
+
+        for metadata in windows_os_meta:
+            location = metadata.get("Location")
+            version = metadata.get("Version")
+            if location and version:
+                return location, version
+
+    except Exception as e:
+        print(f"Unable to get download url: {e}")
+    return None, None
+
+
+def _install_daqmx(download_url, temp_file):
+    """
+    Download and launch NI-DAQmx Driver installation in an interactive mode
+    """
+    try:
+        urllib.request.urlretrieve(download_url, temp_file)
+        subprocess.Popen([temp_file], shell=True)
```

Comment:
`subprocess.Popen` is for when you want to interact with the child process's stdin, stdout, and/or stderr. If you aren't doing that, use a different `subprocess` function such as `subprocess.run`.

Also, this leaks a subprocess, which is why you got a `ResourceWarning` when you demoed this earlier. Using `subprocess.run` will wait for the command to complete, which should fix this leak.